openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -103,20 +112,117 @@ class ParameterList(list):
 
     # TODO: Allow retrieval by `id` as well
 
+    def __init__(self, input_parameter_list=None):
+        """
+        Initialize a new ParameterList, optionally providing a list of ParameterType objects
+        to initially populate it.
+
+        Parameters
+        ----------
+        input_parameter_list: list[ParameterType], default=None
+            A pre-existing list of ParameterType-based objects. If None, this ParameterList
+            will be initialized empty.
+        """
+        super().__init__()
+
+        # We keep track of the last parameter added as this will be
+        # used to set output units during serialization
+        self._last_added_param = None
+        input_parameter_list = input_parameter_list or []
+        # TODO: Should a ParameterList only contain a single kind of ParameterType?
+        for input_parameter in input_parameter_list:
+            self.append(input_parameter)
+            self._last_added_param = input_parameter_list
+
+    @property
+    def last_added_parameter(self):
+        """
+        Get a copy of the last parameter added to this ParameterList. Important
+        during serialization as the last parameter added determines the units used for
+        serializing all other parameters of this type.
```

Comment:
> One alternative solution, is to expose something like a default_serialization_unit property.

I think this is what we do with the `output_units` parameter for `to_dict` in our `ParameterType` and `ParameterHandler` classes.

> I wouldn't expose publicly the internals of how we make the default decision.

Not sure if I totally catch the meaning here. I agree that we shouldn't let users modify `_last_added_param`. Is this referring to something else?

Reply:
I was referring to the public `last_added_parameter` property here, sorry! What I mean is that if in the future we want to change how we select the default units (currently we use the units of the last added parameter type), we'll have to break the public API. If we expose the information of the current default units with a `default_serialization_unit` instead of `last_added_parameter`, the transition to a different default scheme will be mostly harmless.

Reply:
Thanks for pushing this. I forgot that I exposed this as a `@property`, and didn't actually look at the referenced code on your comment. You're totally right, and I was lazy and didn't look closely enough at your feedback. I agree that this shouldn't be a public method. Removed in 409e2ae. 

I'm hesitant to expose `default_serialization_unit`, at this time, since we already support the `output_units` parameter during serialization. I'm going to leave it out for now but would consider adding it in a later PR.